crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -0,0 +1,1154 @@
+								}
+								return bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						for (let foo = 0; foo < 1; foo++) {
+							function doBar(bar) {
+								return bar + foo;
+							}
+						}
+					",
+            None,
+        ),
+        (
+            "
+						let foo = 0;
+						function doFoo() {
+							foo = 1;
+							function doBar(bar) {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo =
+							foo =>
+							bar =>
+							foo + bar;
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = () => {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							return bar => bar;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						const doFoo = foo => {
+							const doBar = bar => {
+								return foo + bar;
+							}
+							return foo;
+						}
+					",
+            None,
+        ),
+        (
+            "
+						function doFoo() {
+							{
+								const foo = 'foo';
+								function doBar(bar) {
+									return bar + foo;
+								}
+							}
+						}
+					",
+            None,
+        ),
+        // eslint-plugin-unicorn allows this
+        // but i think we should report it
+        //      (
+        //          "
+        //  					function doFoo(foo) {
+        //  						{
+        //  							function doBar(bar) {
+        //  								return bar;
+        //  							}
+        //  						}
+        //  						return foo;
+        //  					}
+        //  				",
+        //          None,
+        //      ),
+        (
+            "
+						function doFoo(foo) {
+							function doBar(bar) {
+								foo.bar = bar;
+							}
+							function doZaz(zaz) {
+								doBar(zaz);
+							}
+			
+							doZaz('zaz');
+						};
+					",
+            None,
+        ),
+        (
+            "
+        				function doFoo() {
+        					return function doBar() {};
+        				}
+        			",
+            None,
+        ),
+        (
+            "
+        				function doFoo(Foo) {
+        					function doBar() {
+        						return new Foo();
+        					}
+        					return doBar;
+        				};
+        			",
+            None,
+        ),
+        (
+            "
+        				function doFoo(FooComponent) {
+        					return <FooComponent />;
+        				}
+        			",
+            None,
+        ),
+        // (
+        //     "
+        // 				function doFoo(FooComponent) {
+        // 					function Bar() {
+        // 						return <FooComponent />;
+        // 					}
+        // 					return Bar;
+        // 				};
+        // 			",
+        //     None,
+        // ),
```

Comment:
same here. https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md#

known limitation of the unicorn rule. i think we should report it - thoughts?